geo.py

```diff
@@ -0,0 +1,161 @@
+"""
+    Reynir: Natural language processing for Icelandic
+
+    Processor module to extract entity names & definitions
+
+    Copyright (c) 2018 Miðeind ehf.
+
+       This program is free software: you can redistribute it and/or modify
+       it under the terms of the GNU General Public License as published by
+       the Free Software Foundation, either version 3 of the License, or
+       (at your option) any later version.
+       This program is distributed in the hope that it will be useful,
+       but WITHOUT ANY WARRANTY; without even the implied warranty of
+       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+       GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see http://www.gnu.org/licenses/.
+
+
+    This module contains geography and location-related utility functions.
+
+"""
+
+
+import json
+import re
+from iceaddr import iceaddr_lookup
+from country_list import countries_for_language, available_languages
+
+
+ICELAND_ISOCODE = "IS"
+ICELANDIC_LANG_ISOCODE = "is"
+
+COUNTRY_COORDS_JSONPATH = "resources/country_coords.json"
+
+
+def coords_for_country(iso_code):
+    """ Return coordinates for a given country code """
+    assert len(iso_code) == 2 and iso_code.isupper()
+
+    # Lazy-load data, save as function attribute
+    if not hasattr(coords_for_country, "iso2coord"):
+        with open(COUNTRY_COORDS_JSONPATH) as f:
+            coords_for_country.iso2coord = json.load(f)
+
+    return coords_for_country.iso2coord.get(iso_code)
+
+
+def coords_for_street_name(street_name, placename=None, placename_hints=[]):
+    """ Return coordinates for an Icelandic street name as a tuple """
+
+    addresses = iceaddr_lookup(street_name, placename=placename, limit=100)
+    places = set(a["stadur_nf"] for a in addresses if a.get("stadur_nf"))
+    addr = None
+
+    # Street name only exists in one place
+    if len(places) == 1:
+        addr = addresses[0]
```

Comment:
What about the case where `addresses[0]["stadur_nf"]` is `None` or empty? From the set generator expression it looks like that could be the case.

Reply:
Well spotted!